app/UserService.ts

```diff
@@ -405,39 +426,59 @@ export class UserService extends ChordNode {
     }
   }
 
-  migrateUsersToPredecessorOrSuccessor(migrateToPredecessor: boolean) {
-    if (this.objectIsEmpty(this.userMap)) {
-      return null;
-    }
+  migrateUsersToPredecessor() {
+    if (this.userMapIsEmpty()) return null;
+
+    const client = connect(this.predecessor);
 
-    let usersToMigrate;
-    if (migrateToPredecessor) {
-      usersToMigrate = Object.keys(this.userMap).filter(userId =>
+    Object.keys(this.userMap)
+      .filter(hashedKey =>
```

Comment:
Mmm I guess that this was a bug, as the migration was getting done by userId instead of hashed key. 
If we go back to save users in userMap by userId, we will have to use the metadata to determine which ones we are migrating here.